### PR TITLE
Add `pythoninclude` to `pyodide config` CLI

### PR DESCRIPTION
**Pull request by agriyakhetarpal**
### Description

This PR is based on https://github.com/pyodide/pyodide/issues/4551#issuecomment-2229172277. I thought of adding `python_interp_include_dir`, but that name felt too long, so I switched to `python_include_dir`, and I then noticed that `pythoninclude` already exists as a variable that is getting computed through `Makefile.envs`. So, instead of renaming it from `pythoninclude` to `python_include_dir` and inducing an unneeded breaking change, I've just exposed it verbatim here, such that it is both accessible through the `pyodide config get pythoninclude` command for out-of-tree builds and visible in the list of default variables through the `pyodide config list` command.

It is to be noted that `emscripten_version` is the only variable that seems to be in active use for the `pyodide config` CLI based on a quick [GitHub code search](https://github.com/search?q=%22pyodide+config+get%22&type=code); perhaps this is because it is the only one that has been [documented](https://pyodide.org/en/stable/development/building-and-testing-packages.html#set-up-emscripten) before, so even if we were to rename and switch up a few variables, we should be fine (but I would prefer not to, personally).

The rationale behind this change is that some packages might need to read through the [Python headers](https://github.com/python/cpython/tree/main/Include) for some reason or the other, such as #4936. The Emscripten toolchain already has this directory passed to it through the use of the `-I$(PYTHONINCLUDE)` flag, and there is no need for any package to link against Python itself, so I'm unsure if this is an entirely beneficial patch – please let me know! At least, the change is simple and shouldn't cause any regressions, so those who wish to use this variable for their needs can do so through this PR. The use cases I've thought of so far are when compiled extensions that are defined in a package's `setup.py` rely more on the `setuptools` machinery to pass options to CMake, rather than having CMake do its thing with `FindPython`. However, we never know when or if more packages might benefit from using it, so IMO it feels okay to add.

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry

### CODE REVIEW

**Comment by hoodmane** [comment]
If packages need this in `setup.py` shouldn't they do:
```py
import sysconfig

print(sysconfig.get_config_var("INCLUDEPY"))
```

**Comment by hoodmane** [comment]
Perhaps we should add a way to look up Pyodide's sysconfig variables from the cli?

**Comment by agriyakhetarpal** [comment]
I think so too. Do you mean something like `pyodide config get sysconfig::variable`?

**Comment by agriyakhetarpal** [comment]
> If packages need this in `setup.py` shouldn't they do:
> 
> ```python
> import sysconfig
> 
> print(sysconfig.get_config_var("INCLUDEPY"))
> ```

I think `sysconfig.get_path('include')` and `sysconfig.get_config_var('INCLUDEPY'))` are equivalent – both return the same thing.

**Comment by hoodmane** [comment]
There is no variable called `include` or `INCLUDE` you'll just get `None`. But `sysconfig.get_config_var("INCLUDEDIR")` points up a level. Here are all of the related variables in a native Debian install of Python 3.12:
```
INCLUDEDIR ==> /usr/include
INCLUDEPY ==> /usr/include/python3.12
CONFINCLUDEPY ==> /usr/include/python3.12
INCLDIRSTOMAKE ==> /usr/include /usr/include /usr/include/python3.12 /usr/include/python3.12
```

**Comment by agriyakhetarpal** [comment]
Oh, I think you (might) have misread – it's `sysconfig.get_path()`, it returns the values in `sysconfig.get_paths()`: https://docs.python.org/3/library/sysconfig.html#sysconfig.get_path, which are as follows:

```
{'data': '/opt/homebrew/Caskroom/miniforge/base',
 'include': '/opt/homebrew/Caskroom/miniforge/base/include/python3.10',
 'platinclude': '/opt/homebrew/Caskroom/miniforge/base/include/python3.10',
 'platlib': '/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages',
 'platstdlib': '/opt/homebrew/Caskroom/miniforge/base/lib/python3.10',
 'purelib': '/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages',
 'scripts': '/opt/homebrew/Caskroom/miniforge/base/bin',
 'stdlib': '/opt/homebrew/Caskroom/miniforge/base/lib/python3.10'}

```

**Review by ryanking13 — APPROVED** [review]
Thanks @agriyakhetarpal!

Could you open a same PR in [pyodide/pyodide-build](https://github.com/pyodide/pyodide-build) too, after merging this? Or if this patch isn't required right now, it is okay to close this and open it only in pyodide-build repository as I hope I'll be able to merge #4882 in a few days/weeks.

---

> The rationale behind this change is that some packages might need to read through the [Python headers](https://github.com/python/cpython/tree/main/Include) for some reason or the other, such as https://github.com/pyodide/pyodide/pull/4936

Yes, I agree that exposing this directory would be useful.

> If packages need this in setup.py shouldn't they do:

I think it is also common to get these variables in the CLI. For instance there is a `python3-config` CLI for that ([man page](https://helpmanual.io/man1/python3-config/)). So I think providing a similar way of getting those values for cross-compilation seems reasonable to me.

**Comment by agriyakhetarpal** [comment]
Thanks for the review and the additional context, @ryanking13! I don't think the patch is required right now, and I wish to follow up with @hoodmane's suggestion on looking up Pyodide's sysconfig variables from the CLI, i.e., the ones in `sysconfig.get_paths()` as noted above in https://github.com/pyodide/pyodide/pull/4947#issuecomment-2235795955. Loading them dynamically makes sense to me. I can wait for you to merge #4882, and I'll push these changes to https://github.com/pyodide/pyodide-build after that.

**Pull request closed without merging**
